pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSegmentUploadRestletResource.java

```diff
@@ -327,15 +332,19 @@ private SuccessResponse uploadSegment(FormDataMultiPart multiPart, boolean enabl
   }
 
   private String getZkDownloadURIForURIUpload(String currentSegmentLocationURI, SegmentMetadata segmentMetadata,
-      FileUploadPathProvider provider) throws URISyntaxException, UnsupportedEncodingException {
+      String segmentPersistenceStratey, FileUploadPathProvider provider) throws URISyntaxException, UnsupportedEncodingException {
     String zkDownloadUri;
     if (new URI(currentSegmentLocationURI).getScheme().equals(CommonConstants.Segment.LOCAL_SEGMENT_SCHEME)) {
       zkDownloadUri = ControllerConf.constructDownloadUrl(segmentMetadata.getTableName(), segmentMetadata.getName(),
           provider.getVip());
     } else {
-      LOGGER.info("Using configured data dir {}", _controllerConf.getDataDir());
-      zkDownloadUri = StringUtil.join("/", provider.getBaseDataDirURI().toString(), segmentMetadata.getTableName(),
-          URLEncoder.encode(segmentMetadata.getName(), "UTF-8"));
+      if (CommonConstants.Segment.PERSIST_TO_PINOT_MANAGED_FILESYSTEM.equalsIgnoreCase(segmentPersistenceStratey)) {
+        LOGGER.info("Using configured data dir {}", _controllerConf.getDataDir());
+        zkDownloadUri = StringUtil.join("/", provider.getBaseDataDirURI().toString(), segmentMetadata.getTableName(),
+            URLEncoder.encode(segmentMetadata.getName(), "UTF-8"));
+      } else {
+        zkDownloadUri = currentSegmentLocationURI;
```

Comment:
even if you set this, inside "completeZkOperations," I again get the final segment directory and will move it there later on. Will that break your functionality?

Reply:
hmm, so the downloadUri in zk will be override again ?

Reply:
nope, download uri will be right, but the segment will still be copied to local and copied to the remote place. segment will be in both places